File: BackendVeterinaria/Models/Mascota.cs

using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;
namespace Veterinaria.Models;

public class Mascota
{
    public int MascotaId {get;set;} 
    public int PropietarioId { get; set;}
    public string Nombre {get; set;}
    public int TipoId {get; set;}
    public int RazaId {get; set;}
    public string PesoLb {get; set;} 
    public string Genero {get; set;}

    public virtual Propietario Propietario { get; set;}
 
    public virtual Tipo Tipo {get; set;}
   
    public virtual Raza Raza {get; set;} 
  
}

File: FrontendVeterinaria/src/app/models/mascota.ts
import { Propietario } from "./propietario";
import { Raza } from "./raza"
import { Tipo } from './tipo';

export class Mascota {
    mascotaId?: number
    propietarioId?: number
    nombre?: string
    razaId?: number
    tipoId?: number
    pesoLb?: string
    genero?: string
    raza?: Raza
    tipo?: Tipo
    propietario?: Propietario
}